### PR TITLE
MERGE BASE BACKEND:

### LINKED ISSUE #18: Create a walkthrough for installing the back-end (video)
I've created a video for installing the backend nestjs. (No voice) https://www.youtube.com/watch?v=VenDyTwjirM
here is the transcription : 

    Walkthrought For dev

    So we are inside the back-end folder with nestjs 
    We start to create the image with the docker compose
    
    First one : 
    "docker build -t dev ."
    Can take time, -t is for the name of the image
    Flag -g is designed to install globally and not in the project 
    
    here we are, we've got the image ready to the project
    we can put the next line, but change for dev, we dont need to suppress 
    after the exit of the container.
    docker run -ti --rm -v $(pwd)/app:/app -p 3000:3000 dev
    
    So "docker run -ti -v $(pwd)/app:/app -p 3000:3000 dev"
    We say we want to run in interactive tty mode (-ti) and we want to 
    Bind the volume on the current folder/app to /app inside the container
    
    we bind the port 3000 o0n each side. 
    Finaly we select "dev" images
    
    Folder app is already created.
    
    /app/transcendance is the root of the project nestjs
    
    ./dist is the build of the project, already done,
    
    ./package.json is defined to set-up and install the project.
    
    So here we go. I've already configure the project.
    
    we need to type npm install. 
    
    Previous error is generated because the installer of npm doesn't find package.json
    
    Vulnerability we'll be fixed soon,
    
    Now a folder node_modules is created, and a package-lock.json
    We need to avoid to push it on the git.
    In fact it lock the configuration, so you can't install if file is
    available.
    
    So here we are. if we see this, it is all done for set up project. 
    
    We've got script into package.json
    
    Build will create the build of the app into dist, let's try it.
    Some script doesn't require run keyword. 
    
    
    build : npm run build 
    all right !
    
    
    Considering we are in dev mode .
    So we can do "npm run start:dev" 
        it will watch file change.
        
    pm2 script will come soon. But for dev now it take no importance. 
    
    If we want to check the norme of the code (the quality of the code)
    npm run lint
    
    we can see everithing is okay. except type script version but anyway
    if we've got problem, we'll see something like that.
    
    right XD
    All norme will be threated as error. We ckan discuss about
    it if consistance is not good.
    
    DONT FIX IT MANNUALY, or maybe but no warranty about that.
    
    You've got a plugin in vscode called eslint, I choose this one,
    You've got directly correct code style
    
    
    Allright for the code style. 
    
    Let's run in dev mode our app . 
    
    Our app is correctly launch 
    To test our app we go to the first module created by the nest/cli
    
    
    listen on 3000
    
    AppModule have a controller, and a service. 
    
    controller is for the methods, GET, POST etc... 
    
    services is designed to handle the data.
    
    import is designed to add new modules. We skip it for now. 
    
    It crash because I've got auto-save. 
    
    re-launch dev mode 
    
    Common Case, for now we need to close the process, we can't reuse the port 
    
    Installing htop and added to the dockerfile.
        killing with k and press enter to select SIGTERM 
        
        
    Okay, 
    Service is the data handler. 
    
    We see the controller has the decorator @Get from nestjs feature
    It is more simple than express :) 
    
    The decorator @Controller is design to handle the route
    
    for exemple: 
    @Controller() refer to the root of the server 
        in eg : "localhost:3000/"
        
    We can also do: 
    @Controller("products") refer to "localhost:3000/products/"
    
    The decorator @Get(":id") can have a string. :id is for params of 
        the request. Dynamic :)
    
    but can also be a static 
            For example : @Controller("products") and @Get("promo")
            will refer to "localhost:3000/products/promo"
            
    Finaly it is really easy to create module.
    This module is the default module. 
    Next video will explain how to create a CRUD oparation. 
    
    
    For testing manually the module we can use POSTMAN, it is really easy 
    To dev without jest or else. 
    Be carefull to make documentation. :)
    
    So try the route "/"
    
    We can try something else
    
    That's all for now :)

### PR DESCRIPTION
**Pull request by apayetRun**
Committer: apayet <apayet@localhost.localdomain>

close #18 
create the base of the backend serve with nestjs.
makefile created to assists dev to install their environment.
helper inside the makefile.

Just make makefile all and follow instructions.

### CODE REVIEW

**Comment by apayetRun** [comment]
@AdrianWeaver Petite erreur de ma part, 
new file: backend/app/transcendence/src/socket/SocketEvents.ts
new file: backend/app/transcendence/src/socket/socket.modul.ts
Ne doivent pas etre present sur le PR, c'est une autre feature

**Comment by AdrianWeaver** [comment]
Désolé pour le temps de réponse, I'm back :)
Est-ce que tu as retiré les fichiers que tu ne souhaites pas avoir dans un commit supplémentaire ?

**Review by AdrianWeaver — CHANGES_REQUESTED** [review]
Désolé pour le temps de réponse, I'm back :)

Il faut que la description de ton commit soit plus explicite, qu'est-ce que fais ce commit ? 
Visiblement tu ajoutes une base, qui sert à quoi, qui est organisée comment ?

Inutile de mettre tous les fichiers modifiés dans le commit, c'est pas une information très pertinente dans ce contexte.

C'est une bonne chose d'avancer ! Surtout d'avoir une base de travail, toutefois j'ai l'impression qu'il manque quelque chose avant cette PR qui va à l'encontre de ce qui a été décidé, on a pas déterminé la structure des dossiers.
Ça me parait pas déconnant comme structure, mais on a déterminé ça quelque part ?
Pour que tout le monde puisse organiser son taff et créer les bons fichiers aux bons endroit il faut déterminer où et comment faire.
Quels sont les fichiers "automatiques" ou "default" quels sont les fichiers "custom" etc.

Je n'ai pas la compétence technique pour le moment de faire une review de ts/js.
Toutefois ce que je constate c'est qu'il y a des écarts de norme entre les fichiers.
Notamment sur les {} [ ] en début de nouvelle ligne ou en fin de ligne précédente.
Je vais les flags uns par un pour faciliter la review

File: backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM debian:buster
+
+RUN apt update
+RUN echo yes | apt upgrade
+
+RUN echo yes | apt install nodejs npm curl git htop
```
Comment (AdrianWeaver) [inline backend/Dockerfile:6]: why not use apt-get --yes instead of this sorcery ?

File: backend/app/transcendence/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,327 @@
+module.exports = {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:1]: { on end of line.

File: backend/app/transcendence/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+        "plugin:@typescript-eslint/recommended"
+    ],
+    "overrides":
+    [
+        {
+            "env": {
+                "node": true
+            },
+            "files": [".eslintrc.{js,cjs}"],
+            "parserOptions": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:19]: { on end of line.

File: backend/app/transcendence/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    },
+    "extends":
+    [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/recommended"
+    ],
+    "overrides":
+    [
+        {
+            "env": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:15]: { on end of line.

File: backend/app/transcendence/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+    {
+        // "indent":
+        // [
+        //     "error",
+        //     "tab"
+        // ],
+        // "linebreak-style": [
+        //     "error",
+        //     "unix"
+        // ],
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:44]: Why is this commented ? Can it be removed if not needed?

File: backend/app/transcendence/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+            "error",
+            "double"
+        ],
+        "semi":
+        [
+            "error",
+            "always"
+        ],
+        // custom rules based on 42
+        // POSSIBLE PROBLEMS
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:56]: Can you explicit possible problems?
Is it glitch issues, security issues? Is this something we should do differently and are aware of but this is in order to comply with the subject?

File: backend/app/transcendence/.eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+            "error",
+            {
+                "commentPattern": "break[\\s\\w]*omitted"
+            }
+        ],
+        "no-import-assign": "error",
+        "no-inner-declarations": "error",
+        "no-self-compare": "error",
+        "no-use-before-define": "error",
+        // SUGESTED
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:92]: suggested*

File: backend/app/transcendence/.eslintrc.js [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+        [
+            "error",
+            4
+        ],
+        "max-statements":
+        [
+            "error",
+            4
+        ],
+        // "new-cap": "error",
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:171]: why is this commented? Does it serve a specific purpose?

File: backend/app/transcendence/.eslintrc.js [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+        "brace-style":
+        [
+            "error",
+            "allman"
+        ],
+        // "comma-dangle":
+        // [
+        //     "error",
+        //     "always"
+        // ],
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:222]: Same question, does it serve a purpose to keep this part as commented?

File: backend/app/transcendence/.eslintrc.js [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+        "nonblock-statement-body-position":
+        [
+            "error",
+            "below"
+        ],
+        // "object-curly-newline":
+        // [
+        //     "error",
+        //     "always"
+        // ],
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.eslintrc.js:312]: Is this comment needed?

File: backend/app/transcendence/.gitignore [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+*.launch
+.settings/
+*.sublime-workspace
+
+# IDE - VSCode
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
```
Comment (AdrianWeaver) [inline backend/app/transcendence/.gitignore:35]: Really unsure about this part, why would you allow these files to be pushed?
They are local configs why would you want to push them in this project instead of having them on a separate config?

File: backend/app/transcendence/README.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+## Stay in touch
+
+- Author - [Kamil Myśliwiec](https://kamilmysliwiec.com)
+- Website - [https://nestjs.com](https://nestjs.com/)
+- Twitter - [@nestframework](https://twitter.com/nestframework)
+
+## License
+
+Nest is [MIT licensed](LICENSE).
```
Comment (AdrianWeaver) [inline backend/app/transcendence/README.md:73]: why do we have the readme of nest in our project?

File: backend/app/transcendence/nest-cli.json [hunk 13]
```diff
@@ -0,0 +1,8 @@
+{
+  "$schema": "https://json.schemastore.org/nest-cli",
+  "collection": "@nestjs/schematics",
+  "sourceRoot": "src",
+  "compilerOptions": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/nest-cli.json:5]: { on end of line.

File: backend/app/transcendence/package.json [hunk 14]
```diff
@@ -0,0 +1,73 @@
+{
+  "name": "transcendence",
+  "version": "0.0.1",
+  "description": "",
+  "author": "",
+  "private": true,
+  "license": "UNLICENSED",
+  "scripts": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/package.json:8]: { on end of line.

File: backend/app/transcendence/package.json [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    "start:prod": "node dist/main",
+    "start:pm2": "nest build && pm2 start ./dist/main.js -i max",
+    "lint": "npx eslint \"{src,apps,libs,test}/**/*.ts\"",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:cov": "jest --coverage",
+    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:e2e": "jest --config ./test/jest-e2e.json"
+  },
+  "dependencies": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/package.json:23]: { on end of line.

File: backend/app/transcendence/package.json [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+    "@nestjs/common": "^10.0.0",
+    "@nestjs/core": "^10.0.0",
+    "@nestjs/platform-express": "^10.0.0",
+    "@nestjs/platform-socket.io": "^10.0.3",
+    "@nestjs/websockets": "^10.0.3",
+    "reflect-metadata": "^0.1.13",
+    "rxjs": "^7.8.1",
+    "socket.io": "^4.7.0"
+  },
+  "devDependencies": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/package.json:33]: { on end of line.

File: backend/app/transcendence/package.json [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+    "prettier": "^2.8.8",
+    "source-map-support": "^0.5.21",
+    "supertest": "^6.3.3",
+    "ts-jest": "^29.1.0",
+    "ts-loader": "^9.4.3",
+    "ts-node": "^10.9.1",
+    "tsconfig-paths": "^4.2.0",
+    "typescript": "^5.1.3"
+  },
+  "jest": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/package.json:56]: { on end of line.

File: backend/app/transcendence/package.json [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+    "moduleFileExtensions": [
```
Comment (AdrianWeaver) [inline backend/app/transcendence/package.json:57]: [ on end of line.

File: backend/app/transcendence/package.json [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+      "js",
+      "json",
+      "ts"
+    ],
+    "rootDir": "src",
+    "testRegex": ".*\\.spec\\.ts$",
+    "transform": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/package.json:64]: { on end of line.

File: backend/app/transcendence/package.json [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+      "^.+\\.(t|j)s$": "ts-jest"
+    },
+    "collectCoverageFrom": [
```
Comment (AdrianWeaver) [inline backend/app/transcendence/package.json:67]: [ on end of line.

File: backend/app/transcendence/src/app.controller.ts [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+
+@Controller()
+export class AppController
+{
+	constructor(private readonly appService: AppService)
+	{
+
+	}
+
+	@Get()
```
Comment (AdrianWeaver) [inline backend/app/transcendence/src/app.controller.ts:12]: if I understand correctly, if we are willing to keep the habit of adding `;` at the end of our lines there should be one here, right?

File: backend/app/transcendence/src/main.ts [hunk 22]
```diff
@@ -0,0 +1,15 @@
...
+	const app = await NestFactory.create(AppModule);
+	await app.listen(3000);
+}
+
+bootstrap();
+
+/**
+ * Usefull documentation I've found here 
+ * https://www.thisdot.co/blog/introduction-to-restful-apis-with-nestjs/
+ */
```
Comment (AdrianWeaver) [inline backend/app/transcendence/src/main.ts:15]: This is useful but probably not in the proper place, this belongs to the wiki with some explanations as to why and how it's useful. What does it help us with, what did you understand because of it that you didn't, why do you think we should read it, what do you think it would teach us etc.

File: backend/app/transcendence/src/socket/SocketEvents.ts [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+		this.lastWrite = Date.now();
+	}
+
+	// Disconnect
+	handleDisconnect(client: Socket)
+	{
+		console.log("Client disconnect: ", client.id);
+	}
+
+	// Recevoir un event
```
Comment (AdrianWeaver) [inline backend/app/transcendence/src/socket/SocketEvents.ts:42]: Receive an event

File: backend/app/transcendence/src/socket/socket.modul.ts [hunk 24]
```diff
@@ -0,0 +1,11 @@
+import { Module } from "@nestjs/common";
+import { SocketEvents } from "./SocketEvents";
+
+@Module(
+{
+	providers: [ SocketEvents ]
+})
+export class SocketModule
```
Comment (AdrianWeaver) [inline backend/app/transcendence/src/socket/socket.modul.ts:8]: add new line before export maybe?

File: backend/app/transcendence/test/app.e2e-spec.ts [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+	});
+
+	it("/ (GET)", () =>
+	{
+		return request(app.getHttpServer())
+			.get("/")
+			.expect(200)
+			.expect("Hello World!");
+	});
+});
```
Comment (AdrianWeaver) [inline backend/app/transcendence/test/app.e2e-spec.ts:29]: When do we add `;` and when don't we ? This seems to be some kind of a definition, not a call, so probably no use for a `;` here ?

File: backend/app/transcendence/test/jest-e2e.json [hunk 26]
```diff
@@ -0,0 +1,9 @@
+{
+  "moduleFileExtensions": ["js", "json", "ts"],
+  "rootDir": ".",
+  "testEnvironment": "node",
+  "testRegex": ".e2e-spec.ts$",
+  "transform": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/test/jest-e2e.json:6]: { on end of line.

File: backend/app/transcendence/tsconfig.json [hunk 27]
```diff
@@ -0,0 +1,21 @@
+{
+  "compilerOptions": {
```
Comment (AdrianWeaver) [inline backend/app/transcendence/tsconfig.json:2]: { on end of line.

File: backend/build_dev.sh [hunk 28]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+docker build -t dev .
```
Comment (AdrianWeaver) [inline backend/build_dev.sh:2]: Is it the norm to have a .sh file or should we have a makefile instead?

File: backend/doc.MD [hunk 29]
```diff
@@ -0,0 +1 @@
+ https://blog.appsignal.com/2021/11/10/a-guide-to-load-testing-nodejs-apis-with-artillery.html
```
Comment (AdrianWeaver) [inline backend/doc.MD:1]: As said before, this should be wrapped in some kind of explanation, no one wants to follow links blindly even if this one is quite explicit.

Maybe this belongs in the wiki part of the project with a step by step/getting started guide?

File: backend/load_test/Dockerfile [hunk 30]
```diff
@@ -0,0 +1,14 @@
+FROM debian:buster
+
+RUN apt update
+RUN echo yes | apt upgrade
+
+RUN echo yes | apt install nodejs npm curl git
```
Comment (AdrianWeaver) [inline backend/load_test/Dockerfile:6]: apt-get --yes ?

File: backend/load_test/Dockerfile [hunk 31]
```diff
@@ -0,0 +1,14 @@
...
+
+RUN npm install -g n 
+RUN n latest
+
+RUN npm install -g artillery
+
+CMD ["/bin/bash"]
+# docker run -ti -v $(pwd)/script/:/srv/script load_artillery
```
Comment (AdrianWeaver) [inline backend/load_test/Dockerfile:14]: What is this comment ? What purpose does it serve?

File: backend/load_test/script/config/config.yml [hunk 32]
```diff
@@ -0,0 +1,18 @@
+config:
+  target: "http://172.17.0.2:3000"
```
Comment (AdrianWeaver) [inline backend/load_test/script/config/config.yml:2]: What's this IP address ?
Shouldn't this be in the .env file instead?

File: backend/load_test/script/quick_test.sh [hunk 33]
```diff
@@ -0,0 +1 @@
+artillery quick --count 20 --num 10 http://172.17.0.2:3000/
```
Comment (AdrianWeaver) [inline backend/load_test/script/quick_test.sh:1]: Same here, what's the ip address? Shouldn't this be in the .env instead and used as a variable?

File: backend/walkthrough [hunk 34]
```diff
@@ -0,0 +1,222 @@
+
```
Comment (AdrianWeaver) [inline backend/walkthrough:1]: Ok this is the explanation I needed before!
Thanks!
However this does not belong in this commit as this is a documentation this should be added in a DOC commit.

Or even better, in the project wiki :)

File: backend/Dockerfile (hunk 1 above)
Comment (apayetRun) [reply backend/Dockerfile:6]: As You want, was just send "yes" to the standard Input

File: backend/app/transcendence/.eslintrc.js (hunk 10 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:312]: Was needed for testing the linter, today it is no longer needed

File: backend/app/transcendence/.eslintrc.js (hunk 9 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:222]: We can remove it safely after several usage

File: backend/app/transcendence/.eslintrc.js (hunk 8 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:171]: Yes testing purpose, like other no longer needed

File: backend/app/transcendence/.eslintrc.js (hunk 7 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:92]: Yes Splitted the conf object into two main part, bellow this line it is suggested recommandation for a good code quality

File: backend/app/transcendence/.eslintrc.js (hunk 6 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:56]: below this line, these rules are used to prevent logic error into the code.
each rules documentation for example: https://eslint.org/docs/latest/rules/no-await-in-loop

File: backend/app/transcendence/.eslintrc.js (hunk 5 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:44]: yep, it was deafult rules, can be safely removed

File: backend/app/transcendence/.eslintrc.js (hunk 4 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:15]: Yep default config, will be update

File: backend/app/transcendence/.eslintrc.js (hunk 3 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:19]: default confi, will be update

File: backend/app/transcendence/.eslintrc.js (hunk 2 above)
Comment (apayetRun) [reply backend/app/transcendence/.eslintrc.js:1]: Plain object Affectation must follow this style. (when '=' followed by a '{' or ']'), other case will be at a new line, object definition, function.

File: backend/app/transcendence/.gitignore (hunk 11 above)
Comment (apayetRun) [reply backend/app/transcendence/.gitignore:35]: Oups, these file are generated by nestjs cli, I'm unsure too. Need to check why this inside the common files

File: backend/app/transcendence/README.md (hunk 12 above)
Comment (apayetRun) [reply backend/app/transcendence/README.md:73]: The cli of nestjs create automatically these file, it help onboarding people to get started easyly

File: backend/app/transcendence/nest-cli.json (hunk 13 above)
Comment (apayetRun) [reply backend/app/transcendence/nest-cli.json:5]: yep, automatically generated file, will be fixed

File: backend/app/transcendence/package.json (hunk 14 above)
Comment (apayetRun) [reply backend/app/transcendence/package.json:8]: automatic file generation, will be fix

File: backend/app/transcendence/package.json (hunk 15 above)
Comment (apayetRun) [reply backend/app/transcendence/package.json:23]: automatic file generation by cli of nestjs, will be fixed

File: backend/app/transcendence/package.json (hunk 16 above)
Comment (apayetRun) [reply backend/app/transcendence/package.json:33]: Also automatic file generation, will be fixed

File: backend/app/transcendence/package.json (hunk 17 above)
Comment (apayetRun) [reply backend/app/transcendence/package.json:56]: same automatic file generation

File: backend/app/transcendence/package.json (hunk 18 above)
Comment (apayetRun) [reply backend/app/transcendence/package.json:57]: automatic file generation

File: backend/app/transcendence/package.json (hunk 19 above)
Comment (apayetRun) [reply backend/app/transcendence/package.json:64]: automatic file generation

File: backend/app/transcendence/src/app.controller.ts (hunk 21 above)
Comment (apayetRun) [reply backend/app/transcendence/src/app.controller.ts:12]: Yes we are want to have this habit, except, this is a decorator, identifiable by @. It take part as one statement @Get() getHello(): string.
Decorator must be in new line to gave understandable logic.
We can see in other file other subtility.

Comment (apayetRun) [reply backend/app/transcendence/src/app.controller.ts:12]: Decorator override functionnality.

File: backend/app/transcendence/src/socket/SocketEvents.ts (hunk 23 above)
Comment (apayetRun) [reply backend/app/transcendence/src/socket/SocketEvents.ts:42]: will be fix

Comment (apayetRun) [reply backend/app/transcendence/src/socket/SocketEvents.ts:42]: sorry

File: backend/app/transcendence/src/socket/socket.modul.ts (hunk 24 above)
Comment (apayetRun) [reply backend/app/transcendence/src/socket/socket.modul.ts:8]: This is a decorator, it is part of the statement  @Module({ providers: [ SocketEvents ]}) export class SocketModule

File: backend/app/transcendence/test/app.e2e-spec.ts (hunk 25 above)
Comment (apayetRun) [reply backend/app/transcendence/test/app.e2e-spec.ts:29]: this file is automatically generated.
this statement :
	return request(app.getHttpServer())
		.get("/") // ; not added because we add it to another function of the pipeline
		.expect(200) // if we add ; here, we'll break the logic
		.expect("Hello World!"); // ; added at the end of the pipe

Comment (apayetRun) [reply backend/app/transcendence/test/app.e2e-spec.ts:29]: The linter show us automatically when to use ;

File: backend/app/transcendence/test/jest-e2e.json (hunk 26 above)
Comment (apayetRun) [reply backend/app/transcendence/test/jest-e2e.json:6]: automatic generated files

File: backend/app/transcendence/tsconfig.json (hunk 27 above)
Comment (apayetRun) [reply backend/app/transcendence/tsconfig.json:2]: automatic generated file, will be fix

File: backend/build_dev.sh (hunk 28 above)
Comment (apayetRun) [reply backend/build_dev.sh:2]: yes we should have a makefile.

File: backend/load_test/Dockerfile (hunk 31 above)
Comment (apayetRun) [reply backend/load_test/Dockerfile:14]: was test purpose on creation

File: backend/load_test/script/config/config.yml (hunk 32 above)
Comment (apayetRun) [reply backend/load_test/script/config/config.yml:2]: this is the url address of the back, and yes this is a mistake as it is. Not in the scope of the PR, 
to go deeper in explanation, 
When we use docker compose to orchestrate the application, 
we will use something like this if we want.
networks:
  transcendence:
    name: transcendence
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 172.17.0.0/16
          gateway: 172.17.0.1

File: backend/load_test/script/quick_test.sh (hunk 33 above)
Comment (apayetRun) [reply backend/load_test/script/quick_test.sh:1]: Not in the scope of the PR, same answer of config.yml

File: backend/app/transcendence/.eslintrc.js (hunk 7 above)
Comment (AdrianWeaver) [reply backend/app/transcendence/.eslintrc.js:82]: I only explicited the typo in "suggested" written with one "g"

**Review by Keeghaan — DISMISSED** [review]
Everything works perfectly, thank you for all this wonderful work !

File: backend/app/README.md [hunk 35]
```diff
@@ -0,0 +1,73 @@
...
+$ npm run test:cov
+```
+
+## Support
+
+Nest is an MIT-licensed open source project. It can grow thanks to the sponsors and support by the amazing backers. If you'd like to join them, please [read more here](https://docs.nestjs.com/support).
+
+## Stay in touch
+
+- Author - [Kamil Myśliwiec](https://kamilmysliwiec.com)
```
Comment (AdrianWeaver) [inline backend/app/README.md:67]: I find it weird to include a readme that is made by someone else in the repo.
I think this belongs to the wiki instead.

File: backend/app/src/main.ts [hunk 36]
```diff
@@ -0,0 +1,10 @@
+import { NestFactory } from "@nestjs/core";
+import { AppModule } from "./app.module";
+
+async function bootstrap()
+{
+	const app = await NestFactory.create(AppModule);
+	await app.listen(3000);
+}
+
+bootstrap();
```
Comment (AdrianWeaver) [inline backend/app/src/main.ts:10]: knowing that bootstrap is a library, I think renaming this function would make more sense.

**Review by AdrianWeaver — CHANGES_REQUESTED** [review]
Hey!
Again, great work :)
I tried my very best to understand these files that are beyong my understanding for now.

I commented on some files, suggesting changes, please do let me know what you think of it.

Also the _ in the name is not needed.
I would suggest something like
``MERGE BACKEND BASE:``

File: backend/app/.eslintrc.js [hunk 37]
```diff
@@ -0,0 +1,305 @@
...
+            "error",
+            "double"
+        ],
+        "semi":
+        [
+            "error",
+            "always"
+        ],
+        // custom rules based on 42
+        // POSSIBLE PROBLEMS
```
[the remark this answers is not in the document]
Comment (apayetRun) [reply backend/app/.eslintrc.js:46]: This is classification from https://eslint.org/docs/latest/rules
This comment is here for know what can be safely removed from rules
Possible problems may not be removed. 
Suggested depend on code style. Thats means make a uniform code quality.

File: backend/app/src/main.ts (hunk 36 above)
Comment (apayetRun) [reply backend/app/src/main.ts:10]: I see bootstrap staff make a good communication.
This is the nestjs's function architecture.
Here boostrap means (Dictionnary's definition) : a technique of loading a program into a computer by means of a few initial instructions which enable the introduction of the rest of the program from an input device.

File: backend/app/.gitignore [hunk 38]
```diff
@@ -0,0 +1,35 @@
...
+*.launch
+.settings/
+*.sublime-workspace
+
+# IDE - VSCode
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
```
[the remark this answers is not in the document]
Comment (apayetRun) [reply backend/app/.gitignore:35]: We can see it have only the json files, all stuff is discarded at line 31. We've got only config files (.json). I think in other team configuration, it'll make sense to have it. Our workflow depends not on vscode. So we can safely comment it. Anyway I try to understand why nestjs team choose this config. And they take us by the hand. Example launch.json give team mate one way to launch the app from vscode; extensions.json will have recommendation extension for the project (can lead to conflict indeed).

File: backend/app/README.md (hunk 35 above)
Comment (apayetRun) [reply backend/app/README.md:67]: Sorry, I just provided the code as it is to avoid too many change from original template.
We can safely remove this file.

File: backend/app/.eslintrc.js [hunk 39]
```diff
@@ -0,0 +1,305 @@
...
+            "error",
+            {
+                "commentPattern": "break[\\s\\w]*omitted"
+            }
+        ],
+        "no-import-assign": "error",
+        "no-inner-declarations": "error",
+        "no-self-compare": "error",
+        "no-use-before-define": "error",
+        // SUGESTED
```
[the remark this answers is not in the document]
Comment (apayetRun) [reply backend/app/.eslintrc.js:82]: corrected

**Review by Keeghaan — APPROVED** [review]

**Review by nicolkaver — APPROVED** [review]
All works as expected, thank you very much for the efforts you put into making of these files !!

File: backend/Dockerfile [hunk 40]
```diff
@@ -0,0 +1,16 @@
+FROM debian:buster
+
+RUN apt update && apt upgrade -y
+
+RUN apt install -y nodejs npm curl git htop
+
+RUN npm install -g n 
+RUN n latest
```
[the remark this answers is not in the document]
Comment (AdrianWeaver) [reply backend/Dockerfile:8]: What do you think about this one @apayetRun ?

**Review by AdrianWeaver — COMMENTED** [review]
I added a comment on a possible error or error-prone code in the Dockerfile.

I think we can still merge this and add a patch later.

File: backend/Dockerfile (hunk 40 above)
[the remark this answers is not in the document]
Comment (apayetRun) [reply backend/Dockerfile:8]: Yep, this is the starter pack :) the issue #30 will do the job.

**Review by AdrianWeaver — APPROVED** [review]
